resources/js/APIRequest.vue

```diff
@@ -0,0 +1,121 @@
+<script>
+    export default {
+        props: {
+            method: {
+                type: String,
+                required: true,
+            },
+            destination: {
```

Comment:
Rename to url and remove the prefix in the method so it can be used with other endpoints?